ports/psoc6/mbedtls/mbedtls_config.h

```diff
@@ -868,39 +869,154 @@
 
 #endif /* DISABLE_MBEDTLS_ACCELERATION */
 
+#endif /* MBEDTLS_USER_CONFIG_HEADER */
 
+#ifndef PSOC6_MPY_MBEDTLS_USER_CONFIG_HEADER
+#define PSOC6_MPY_MBEDTLS_USER_CONFIG_HEADER
 
-// added by JB
-
-// make sure to comment out lines 1623 - 1629 in modlwip.c JB
-
-// this file was copied from   boards/mtb_shared/wifi-core-freertos-lwip-mbedtls/release-v1.0.0/configs/mbedtls_user_config.h
-// further configs can be found at : boards/mtb_shared/mbedtls/mbedtls-2.25.0/configs/
+// Used to add MPY settings to MTB settings in file "boards/mtb_shared/wifi-core-freertos-lwip-mbedtls/release-v1.0.0/configs/mbedtls_user_config.h"
+// such that both use an identical set of options.
+// Further MTB configs can be found at : boards/mtb_shared/mbedtls/mbedtls-2.25.0/configs/
 
-// this is the content of old lwipopts.h JB :
-// // Set mbedtls configuration
+// Set mbedtls configuration
 #define MBEDTLS_ECP_NIST_OPTIM
 #define MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
 
-// // Enable mbedtls modules
+// // // Enable mbedtls modules
 #define MBEDTLS_GCM_C
 #define MBEDTLS_HAVE_TIME
 #define MBEDTLS_HAVE_TIME_DATE
 
 // // #define MBEDTLS_NO_PLATFORM_ENTROPY
 #define MBEDTLS_CIPHER_MODE_CTR
+#define MBEDTLS_CIPHER_MODE_CBC
 
-// // // Time hook
-// // #include <time.h>
-// // time_t psoc6_rtctime_seconds(time_t *timer);
-// // #define MBEDTLS_PLATFORM_TIME_MACRO rp2_rtctime_seconds
+// Time hook
+// #include <time.h>
+// time_t psoc6_rtctime_seconds(time_t *timer);
+// #define MBEDTLS_PLATFORM_TIME_MACRO psoc6_rtctime_seconds
 
 // Set MicroPython-specific options.
 #define MICROPY_MBEDTLS_CONFIG_BARE_METAL (1)
 
 // Include common mbedtls configuration.
+#ifdef MICROPY_CONFIG_ROM_LEVEL
 #include "extmod/mbedtls/mbedtls_config_common.h"
+#endif
 
-#undef MBEDTLS_ENTROPY_HARDWARE_ALT
+// #undef MBEDTLS_ENTROPY_HARDWARE_ALT
 
-#endif /* MBEDTLS_USER_CONFIG_HEADER */
+#endif /* PSOC6_MPY_MBEDTLS_USER_CONFIG_HEADER */
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
```

Comment:
Maybe this header is merged at the top with our own in the next refactor round.

Reply:
yes, we should clean up properly this config file. Currently is just all of them mpy, mtb, merged.

Reply:
done!